cmd/erasure-metadata.go

```diff
@@ -204,10 +204,23 @@ func (fi FileInfo) TransitionInfoEquals(ofi FileInfo) bool {
 
 // MetadataEquals returns true if FileInfos Metadata maps are equal, false otherwise.
 func (fi FileInfo) MetadataEquals(ofi FileInfo) bool {
-	if len(fi.Metadata) != len(ofi.Metadata) {
-		return false
-	}
+	tierFVIDKey := ReservedMetadataPrefixLower + tierFVID
+	tierFVMarkerKey := ReservedMetadataPrefixLower + tierFVMarker
+
```

Comment:
Since we no longer compare the lengths we must also check that all keys from ofi is present in fi - otherwise we risk false positives - for example  everything is equal to an empty `fi.Metadata`.

Reply:
why? @klauspost we get the `fi` to be LatestMeta() after quorum calculation.

Reply:
I mean we don't have to do reverse because we already know the `fi` is the good copy.

Reply:
I actually took a different approach.. PTAL